iot-hub/Samples/service/RoleBasedAuthenticationSample/Parameters.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using CommandLine;
+using Microsoft.Azure.Devices;
+
+namespace RoleBasedAuthenticationSample
+{
+    /// <summary>
+    /// Parameters for the application.
+    /// </summary>
+    internal class Parameters
+    {
+        [Option(
+            'h',
+            "HostName",
+            Required = true,
+            HelpText = "The IoT hub host name.")]
+
+        public string HostName { get; set; }
+
+        [Option(
+           'd',
+           "DeviceId",
+           Required = true,
+           HelpText = "The IoT hub device to send a message to.")]
+
+        public string DeviceId { get; set; }
+
+        [Option(
+            't',
+            "TransportType",
+            Default = TransportType.Amqp,
+            Required = false,
+            HelpText = "The transport to use to communicate with the IoT Hub. Possible values include Amqp and Amqp_WebSocket_Only.")]
+        
+        public TransportType TransportType { get; set; }
+
+        [Option(
+            'i',
+            "clientId",
+            Required = true,
+            HelpText = "Client Id of the application Id to login, or the application Id used to log the user in.")]
+        public string ClientId { get; set; }
+
+        [Option(
+            't',
+            "tenantId",
+            Required = true,
+            HelpText = "Application tenant Id")]
```

Comment:
Is this the AAD app's tenant ID? should we be a bit more descriptive?

Reply:
yes it is. Same as above. Let me find better description for all the AAD stuff and add.